File: QuanLyKhachSanCli/src/models/ThuePhong.ts
import { BangGia } from "@/models/BangGia";
import { HoaDon } from "@/models/HoaDon";
import { Phong } from "@/models/Phong";

export interface ThuePhong {
    ThuePhongID: number;
    PhongID: number;
    HoaDonID: number;
    
    BangGiaID: number;
    BangGia?: BangGia;
    HoaDon?: HoaDon;
    Phong?: Phong;
}

File: QuanLyKhachSanApp/QuanLyKhachSanApp/Models/ThuePhong.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace QuanLyKhachSanApp.Models
{
    [Table("ThuePhong")]
    public class ThuePhong
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Required]
        public int ThuePhongID { get; set; }

        public int? PhongID { get; set; }

        public int? HoaDonID { get; set; }


        public int? BangGiaID { get; set; }

        [ForeignKey("BangGiaID")]
        public virtual BangGia BangGia { get; set; }

        [ForeignKey("HoaDonID")]
        public virtual HoaDon HoaDon { get; set; }

        [ForeignKey("PhongID")]
        public virtual Phong Phong { get; set; }

    }
}
